src/main/java/appeng/tile/storage/TileDrive.java

```diff
@@ -156,6 +182,7 @@ public boolean isPowered() {
     public boolean readFromStream_TileDrive(final ByteBuf data) {
         final int oldState = this.state;
         this.state = data.readInt() & STATE_MASK;
+        this.type = data.readInt();
```

Comment:
Create variable `oldType` before reading NBT data and compare it with new value in return statement (like it's done in `TileChest:readFromStream_TileChest`).